UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetWhenAioIsEnabled.cs

```diff
@@ -0,0 +1,42 @@
+﻿using NUnit.Framework;
+using System.IO;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.FunctionalTests.Helper;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.FunctionalTests
+{
+    [TestFixture]
+    class ExchangeSetWhenAioIsEnabled
+    {
+        private string FssJwtToken { get; set; }
+        private TestConfiguration Config { get; set; }
+        private string DownloadedFolderPath { get; set; }
+
+
+        [OneTimeSetUp]
+        public async Task SetupAsync()
+        {
+            Config = new TestConfiguration();
+            AuthTokenProvider authTokenProvider = new AuthTokenProvider();
+            FssJwtToken = await authTokenProvider.GetFssToken();
+        }
+
+        [Test]
+        [Category("SmokeTest")]
+        public async Task WhenIDownloadAioZipExchangeSet_ThenAReadmeTxtFileIsAvailableAsync()
+        {
+            foreach (string batchId in Config.AIOConfig.AioExchangeSetBatchIds)
+            {
+                DownloadedFolderPath = await FileContentHelper.DownloadAndExtractAioZip(FssJwtToken, batchId);
+
+                bool checkFile = FssBatchHelper.CheckforFileExist(Path.Combine(DownloadedFolderPath, Config.ExchangeSetEncRootFolder), Config.ExchangeReadMeFile);
+                Assert.IsTrue(checkFile, $"{Config.ExchangeReadMeFile} File not Exist in the specified folder path : {Path.Combine(DownloadedFolderPath, Config.ExchangeSetEncRootFolder)}");
+
+                //Verify README.TXT file content
+                FileContentHelper.CheckAioReadMeTxtFileContent(Path.Combine(DownloadedFolderPath, Config.ExchangeSetEncRootFolder, Config.ExchangeReadMeFile));
+
+                FileContentHelper.DeleteDirectory(Config.AIOConfig.AioExchangeSetFileName);
+            }
+        }
```

Comment:
just another query regarding these. would there need to be additional tests to cover 

1. file not present
2. file present but is corrupted.
3. is there anything restricting large files

Reply:
Hi Ray,

The actual functionality will be developed part of the future PBIs, and the scenario will get covered in those. As part of this PBI we have used the static batches. I hope that's fine.

Regards,
Mayuresh Gawde